sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SymbolicExecution/CSharp/CSharpExplodedGraph.cs

```diff
@@ -520,10 +525,20 @@ protected override void VisitInstruction(ExplodedGraphNode node)
                     break;
 
                 case SyntaxKindEx.ConstantPattern:
+                case SyntaxKindEx.RecursivePattern:
                     // The 0 in 'case 0 when ...'
                     // Do nothing
                     break;
 
+                case SyntaxKindEx.DeclarationExpression:
+                    // e.g.: dictionary.TryGetValue(key, out TValue value);
+                    // Do nothing
+                    break;
+
+                case SyntaxKindEx.TupleExpression:
```

Comment:
Last note: I would expect to see the changes in `CSharpExplodedGraph` to be reflected in `ExplodedGraphTests`. As currently, we only check that these syntaxes do not fail for rules we expect, but we don't check that the exploded graph behaves as expected